### DISCUSSION TITLE
Does @model_validator support async methods?

### DISCUSSION THREAD

**Discussion by skynavga** [pydantic/pydantic#9888] [Question] [answered]
For example, I would like to do something like:

```python
from typing import Any, Self
from my_async_validator import async_validate
class MyModel:
    field1: Any
    field2: Any
    @model_validator(mode='after')
    async def validate(self) -> Self:
        await async_validate(self.field1, self.field2)
        return self
```

however, `mypy` is reporting an error on the `@model_validator` line (6) as follows:

``error: Cannot infer function type argument [misc]``

**Comment by skynavga** [comment, marked as answer]
I was able to resolve this by changing the model validator method to synchronous, then using the asyncio event loop to run the asynchronous code:

```python
from asyncio
from typing import Any, Self
from my_async_validator import async_validate
class MyModel:
    field1: Any
    field2: Any
    @model_validator(mode='after')
    def validate(self) -> Self:
        loop = asyncio.get_event_loop()
        loop.run_until_complete(async_validate(self.field1, self.field2))
        return self
```

**Reply by luqijun** [reply]
> I was able to resolve this by changing the model validator method to synchronous, then using the asyncio event loop to run the asynchronous code:
> 
> ```python
> from asyncio
> from typing import Any, Self
> from my_async_validator import async_validate
> class MyModel:
>     field1: Any
>     field2: Any
>     @model_validator(mode='after')
>     def validate(self) -> Self:
>         loop = asyncio.get_event_loop()
>         loop.run_until_complete(async_validate(self.field1, self.field2))
>         return self
> ```

this not work for me, i got the the error:

```python
  File "uvloop/loop.pyx", line 1511, in uvloop.loop.Loop.run_until_complete
    raise
  File "uvloop/loop.pyx", line 1504, in uvloop.loop.Loop.run_until_complete
    self.run_forever()
  File "uvloop/loop.pyx", line 1377, in uvloop.loop.Loop.run_forever
    self._run(mode)
  File "uvloop/loop.pyx", line 518, in uvloop.loop.Loop._run
    raise RuntimeError('this event loop is already running.')

RuntimeError: this event loop is already running.
```
